Ai Pin/Views/Notes/NotesEditView.swift

```diff
@@ -0,0 +1,63 @@
+import SwiftUI
+import OSLog
+
+struct NotesEditView: View {
+    @Environment(NavigationStore.self) private var navigationStore
+    @Environment(\.dismiss) private var dismiss
+    @State private var triggerSaveHatic = false
+    @State private var triggerCancelHatic = false
+    @State var noteId: String
+    @State var note: Note
```

Comment:
you cannot instantiate @State from external sources. This is incorrect and leads to undefined behavior. What would be better would be for the note to be an observable of some sort, which I intend to do in SwiftDtcata. 

That said, to fix this for now you should use a custom initializer to store the note statically then set state in onAppear.